src/kernels/MIOpenPadConstant.cpp

```diff
@@ -0,0 +1,190 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <cstddef>
+#include <cstdio>
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_runtime.h>
+#include <hip/hip_fp16.h>
+#endif
+
+#include "float_types.h"
+#include "tensor_view_5d.hpp"
+
+template <typename TI, typename TO>
+__device__ void padconstantfwd(const TI* __restrict__ x,
+                               TO* __restrict__ y,
+                               const tensor_view_5d_t x_tv,
+                               const tensor_view_5d_t y_tv,
+                               const padding_5d_t padding,
+                               const size_t output_size,
+                               TO value)
+{
+    TO padding_value   = CVT_ACCUM2FLOAT(value);
```

Comment:
There is no need to use CVT_ACCUM2FLOAT because the value type is already TO.

Reply:
I will double check but I believe it was there because in case of bfp16 being used, that would be necessary. fp32 and fp16 will just skip over it

Reply:
I’ve removed the unnecessary macro.